app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java

```diff
@@ -69,32 +66,30 @@ public void checkToken() {
     }
 
     public void sendPost(String title, String body) {
-        Login login = new Login(title.trim(), body.trim());
-        ApiInterface apiService = ApiClient.getClient().create(ApiInterface.class);
-        Call<LoginResponse> call = apiService.login(login);
-        call.enqueue(new Callback<LoginResponse>() {
-            @Override
-            public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
-                if (response.isSuccessful()) {
-                    String accessToken = response.body().getAccessToken();
-                    Toast.makeText(getApplicationContext(), "Success! ", Toast.LENGTH_LONG).show();
-                    TOKEN = accessToken;
-                } else {
-                    Toast.makeText(getApplicationContext(), "Error Occured " + response.code(), Toast.LENGTH_LONG).show();
-                    Log.d("harsimarSingh", "Error " + response.code() + " Error body " + response.errorBody());
-                }
-                progressDialog.cancel();
-                checkToken();
-            }
+        Login login=new Login(title.trim(),body.trim()) ;
 
-            @Override
-            public void onFailure(Call<LoginResponse> call, Throwable t) {
-                Toast.makeText(getApplicationContext(), "Error contacting API.", Toast.LENGTH_LONG).show();
-                Log.d("harsimarSingh", "Failure " + t.toString());
-                TOKEN = null;
-                checkToken();
-                progressDialog.cancel();
-            }
-        });
+        compositeDisposable.add(ApiClient.getOpenEventAPI().login(login)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(response -> {
+                        Log.d("Message","Token Saved / Logged In Succesfully !");
+                        String accessToken = response.getAccessToken();
+                        Toast.makeText(getApplicationContext(), "Success!" , Toast.LENGTH_LONG).show();
+                        TOKEN = accessToken;
+                        progressDialog.cancel();
+                        checkToken();
```

Comment:
What if the user types an incorrect email and password ? Please add all the logic that you are removing back here. Plus pull the latest changes code has been updated

Reply:
Sure